src/components/search/BreadCrumbs.js

```diff
@@ -2,47 +2,126 @@ import React from 'react';
+    const datePubmedAdded = useSelector(state => state.search.datePubmedAdded);
+    const datePubmedModified = useSelector(state => state.search.datePubmedModified);
+    const datePublished = useSelector(state => state.search.datePublished);
+    const dateCreated = useSelector(state => state.search.dateCreated);
     const dispatch = useDispatch();
 
+    const handleRemoveDate = (dateType) => {
+        switch (dateType) {
+            case 'datePubmedAdded':
+                dispatch(removeDatePubmedAdded());
+                break;
+            case 'datePubmedModified':
+                dispatch(removeDatePubmedModified());
+                break;
+            case 'datePublished':
+                dispatch(removeDatePublished());
+                break;
+            case 'dateCreated':
+                dispatch(removeDateCreated());
+                break;
+            default:
+                break;
+        }
+        dispatch(searchReferences());
+    };
+
+    const handleRemoveFacet = (facet, value) => {
+        dispatch(removeFacetValue(facet, value));
+        dispatch(searchReferences());
+    };
+
     return (
         <Container fluid>
             <Row>
-                <Col style={{textAlign: "left"}}>
-                    {Object.entries(searchFacetsValues).map(([facet, values]) =>
-                        <span key={facet + "_group"}>
-                            {values.map(value =>
-                                <span key={value + "_breadcrumb"}>
-                                    <Button variant="outline-secondary">{(RENAME_FACETS.hasOwnProperty(facet) ? RENAME_FACETS[facet] : facet.replace('.keyword', '').replaceAll('_', ' ')) + ": " + value} &nbsp;
-                                        <RiCloseFill onClick={() => dispatch(removeFacetValue(facet, value))}/>
-                                    </Button>&nbsp;&nbsp;
-                                </span>)}
+                <Col style={{ textAlign: "left" }}>
+                    {Object.entries(searchFacetsValues).map(([facet, values]) => (
+                        Array.isArray(values) ? values.map(value => (
+                            <span key={facet + value + "_breadcrumb"}>
+                                <Button variant="outline-secondary">
+                                    {(RENAME_FACETS.hasOwnProperty(facet) ? RENAME_FACETS[facet] : facet.replace('.keyword', '').replaceAll('_', ' ')) + ": " + value} &nbsp;
+                                    <RiCloseFill onClick={() => handleRemoveFacet(facet, value)} />
+                                </Button>&nbsp;&nbsp;
+                            </span>
+                        )) : (
+                            <span key={facet + "_breadcrumb"}>
+                                <Button variant="outline-secondary">
+                                    {`${facet.replace('datePubmedAdded', 'Date Added to PubMed')
```

Comment:
Here all dates are hardcoded. Is there a way to make the code more flexible and derive the names from the facets?

Reply:
fixed